astropy/io/fits/hdu/table.py

```diff
@@ -537,27 +537,28 @@ def _verify(self, option='warn'):
         """
 
         errs = super()._verify(option=option)
-        if not (isinstance(self._header[0], str) and
-                self._header[0].rstrip() == self._extension):
-
-            err_text = 'The XTENSION keyword must match the HDU type.'
-            fix_text = f'Converted the XTENSION keyword to {self._extension}.'
-
-            def fix(header=self._header):
-                header[0] = (self._extension, self._ext_comment)
-
-            errs.append(self.run_option(option, err_text=err_text,
-                                        fix_text=fix_text, fix=fix))
-
-        self.req_cards('NAXIS', None, lambda v: (v == 2), 2, option, errs)
-        self.req_cards('BITPIX', None, lambda v: (v == 8), 8, option, errs)
-        self.req_cards('TFIELDS', 7,
-                       lambda v: (_is_int(v) and v >= 0 and v <= 999), 0,
-                       option, errs)
-        tfields = self._header['TFIELDS']
-        for idx in range(tfields):
-            self.req_cards('TFORM' + str(idx + 1), None, None, None, option,
-                           errs)
+        if (len(self._header) > 1):
+            if not (isinstance(self._header[0], str) and
+                    self._header[0].rstrip() == self._extension):
+
+                err_text = 'The XTENSION keyword must match the HDU type.'
+                fix_text = f'Converted the XTENSION keyword to {self._extension}.'
+
+                def fix(header=self._header):
+                    header[0] = (self._extension, self._ext_comment)
+
+                errs.append(self.run_option(option, err_text=err_text,
+                                            fix_text=fix_text, fix=fix))
+
+            self.req_cards('NAXIS', None, lambda v: (v == 2), 2, option, errs)
+            self.req_cards('BITPIX', None, lambda v: (v == 8), 8, option, errs)
+            self.req_cards('TFIELDS', 7,
+                           lambda v: (_is_int(v) and v >= 0 and v <= 999), 0,
+                           option, errs)
+            tfields = self._header['TFIELDS']
+            for idx in range(tfields):
+                self.req_cards('TFORM' + str(idx + 1), None, None, None, option,
+                               errs)
```

Comment:
I don't understand why this change ? (It seems to give the same error in the end).

Reply:
I have open two fake PRs in my repo to explain this.
The [first one](https://github.com/cmarmo/astropy/pull/2) only contains the new ``clear()`` function.
The specific test is [failing](https://github.com/cmarmo/astropy/runs/6085659937?check_suite_focus=true#step:6:1455).
The [second PR](https://github.com/cmarmo/astropy/pull/3) adds the control on the header length and [passes](https://github.com/cmarmo/astropy/runs/6085908385?check_suite_focus=true#step:6:1911).
Also the checks on the table specific keywords should be in the `if` compound otherwise the test fails with a different error.
My understanding is that ``CompImageHDU`` inherits from ``BinTableHDU`` and this forces the header to be verified in two ``verify()`` functions while ImageHDU headers are verified only once.... but I know I have a partial understanding of the whole picture....